### PR TITLE
Disable threading in ImplicitEulerExtrapolation

### PR DESCRIPTION
**Pull request by saurabhkgp21**

### CODE REVIEW

File: src/perform_step/extrapolation_perform_step.jl [hunk 1]
```diff
@@ -344,9 +371,8 @@ function perform_step!(integrator,cache::ImplicitEulerExtrapolationConstantCache
 
   max_order = min(size(T)[1], cur_order+1)
 
-  let max_order=max_order, dt=dt, integrator=integrator, cache=cache, repeat_step=repeat_step,
-    uprev=uprev, T=T
-    Threads.@threads for i in 1:2
+  if integrator.alg.threading==false
```
Comment (devmotion) [inline src/perform_step/extrapolation_perform_step.jl]:
```suggestion
  if !integrator.alg.threading
```

File: src/perform_step/extrapolation_perform_step.jl [hunk 2]
```diff
@@ -369,6 +395,33 @@ function perform_step!(integrator,cache::ImplicitEulerExtrapolationConstantCache
...
+              u_tmp = u_tmp + k
+              k_copy = f(u_tmp, p, t+j*dt_temp)
+          end
+          T[index,1] = u_tmp
+        end
+      end
+
+      for i=2:max_order
+        for j=2:i
+          T[i,j] = ((2^(j-1))*T[i,j-1] - T[i-1,j-1])/((2^(j-1)) - 1)
```
Comment (devmotion) [inline src/perform_step/extrapolation_perform_step.jl]: I guess, `max_order` is usually quite small, but still we should try to access `T` in memory order along columns to improve performance, see https://docs.julialang.org/en/v1.2/manual/performance-tips/#Access-arrays-in-memory-order,-along-columns-1.

File: src/perform_step/extrapolation_perform_step.jl [hunk 3]
```diff
@@ -253,33 +253,60 @@ function perform_step!(integrator,cache::ImplicitEulerExtrapolationCache,repeat_
...
+              @.. u_tmps[Threads.threadid()] = u_tmps[Threads.threadid()] + k_tmps[Threads.threadid()]
+              f(k_tmps[Threads.threadid()], u_tmps[Threads.threadid()],p,t+j*dt_temp)
+          end
+
+          @.. T[index,1] = u_tmps[Threads.threadid()]
+        end
+      end
+      for i in 2:max_order
+        for j in 2:i
+          @.. T[i,j] = ((2^(j-1))*T[i,j-1] - T[i-1,j-1])/((2^(j-1)) - 1)
```
Comment (devmotion) [inline src/perform_step/extrapolation_perform_step.jl]: As below, accessing `T` along columns should be faster, see https://docs.julialang.org/en/v1.2/manual/performance-tips/#Access-arrays-in-memory-order,-along-columns-1.

File: src/perform_step/extrapolation_perform_step.jl [hunk 4]
```diff
@@ -253,33 +253,60 @@ function perform_step!(integrator,cache::ImplicitEulerExtrapolationCache,repeat_
...
+            f(k_tmps[1], u_tmps[1],p,t+j*dt_temp)
         end
 
-        @.. T[index,1] = u_tmps[Threads.threadid()]
+        @.. T[index,1] = first(u_tmps)
       end
     end
     for i in 2:max_order
       for j in 2:i
         @.. T[i,j] = ((2^(j-1))*T[i,j-1] - T[i-1,j-1])/((2^(j-1)) - 1)
```
Comment (devmotion) [inline src/perform_step/extrapolation_perform_step.jl]: Same comment as below applies here.

**Comment by ChrisRackauckas** [comment]
We can clean up some of the performance things later. I think it's more important to finish the project and get implicit midpoint done, and we can optimize later

**Comment by ChrisRackauckas** [comment]
Are the other implicit ones fine?

**Comment by saurabhkgp21** [comment]
I haven't open PR for other implicit methods yet.

**Comment by devmotion** [comment]
There's a test error due to ImplicitEulerExtrapolation: https://travis-ci.org/JuliaDiffEq/OrdinaryDiffEq.jl/jobs/575871642 (the other one is due to the downstream tests and unrelated)

**Comment by ChrisRackauckas** [comment]
There's something wrong with the sizes there. Something is vec'd and something isn't

**Comment by devmotion** [comment]
I've updated the PR with the latest changes of master. Unfortunately the changes are not tested on Gitlab, but locally on my computer the extrapolation tests passed.

**Pull request merged**